akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ReplicatedEntityProvider.scala

```diff
@@ -55,15 +61,86 @@ object ReplicatedEntityProvider {
       (settingsPerReplicaFactory(typeKey, replicaId), typeName)
     }.toVector, directReplication = true)
   }
+
+  /**
+   * Scala API
+   *
+   * Create a [[ReplicatedEntityProvider]] that uses the defaults for [[Entity]] when running in
+   * ClusterSharding. A replica will be run per data center.
+   */
+  def perDataCenter[M: ClassTag, E](typeName: String, allReplicaIds: Set[ReplicaId])(
+      create: ReplicationId => Behavior[M]): ReplicatedEntityProvider[M] = {
+    apply(typeName, allReplicaIds) { (typeKey, replicaId) =>
+      ReplicatedEntity(replicaId, Entity(typeKey) { entityContext =>
+        create(ReplicationId.fromString(entityContext.entityId))
+      }.withDataCenter(replicaId.id))
+    }
+  }
+
+  /**
+   * Scala API
+   *
+   * Create a [[ReplicatedEntityProvider]] that uses the defaults for [[Entity]] when running in
+   * ClusterSharding. The replicas in allReplicaIds should be roles used by nodes. A replica for each
+   * entity will run on each role.
+   */
+  def perRole[M: ClassTag, E](typeName: String, allReplicaIds: Set[ReplicaId])(
+      create: ReplicationId => Behavior[M]): ReplicatedEntityProvider[M] = {
+    apply(typeName, allReplicaIds) { (typeKey, replicaId) =>
+      ReplicatedEntity(replicaId, Entity(typeKey) { entityContext =>
+        create(ReplicationId.fromString(entityContext.entityId))
+      }.withRole(replicaId.id))
+    }
+  }
+
+  /**
+   * Java API
+   *
+   * Create a [[ReplicatedEntityProvider]] that uses the defaults for [[Entity]] when running in
+   * ClusterSharding. A replica will be run per data center.
+   */
+  def createPerDataCenter[M](
+      messageClass: Class[M],
+      typeName: String,
+      allReplicaIds: JSet[ReplicaId],
+      createBehavior: akka.japi.function.Function[ReplicationId, Behavior[M]]): ReplicatedEntityProvider[M] = {
```

Comment:
do we have a reason for using japi here instead of `java.util.function`? it's not supposed to throw exceptions and I don't think variance is important here?

Reply:
but I see that Sharding Entity is also using japi

Reply:
I don't think we have a reason either way, i'll switch it.